IPython/core/interactiveshell.py

```diff
@@ -922,18 +922,7 @@ def init_virtualenv(self):
             p = Path(os.readlink(p))
             paths.append(p.resolve())
 
-        # In Cygwin paths like "c:\..." and '\cygdrive\c\...' are possible
-        if str(p_venv).startswith("\\cygdrive"):
-            p_venv = Path(str(p_venv)[11:])
-        elif len(str(p_venv)) >= 2 and str(p_venv)[1] == ":":
-            p_venv = Path(str(p_venv)[2:])
```

Comment:
I'm a little bit lost here - is it really the case that pathlib knows that `Path('\cygdrive\c\...')` == `Path('C:\...')`? :exploding_head: I don't have an environment to check this right now.

Reply:
```py
In [1]: from pathlib import PureWindowsPath

In [2]: PureWindowsPath('C:\\foo') == PureWindowsPath('\\cygdrive\\foo')
Out[2]: False
```

Reply:
I think this has been fixed in pypa/virtualenv#1970. But if it was, reducing `p_venv` here by `11` characters should've made cygwin users always receive this warning :thinking:. Gotta investigate.

Reply:
Maybe this clears up a little bit of confusion.
![image](https://user-images.githubusercontent.com/41439841/133877534-d1cd5022-64fa-4517-a0c9-727c8b65824a.png)

Reducing p_venv by 11 removes the cygwin part. I am not sure if I am understanding the problem entirely here.

Reply:
Also, I have cygwin setup. So I would be happy to test out any changes if required!

Reply:
> Reducing p_venv by 11 removes the cygdrive part. I am not sure if I am understanding the problem entirely here.

Yes, and so I was confused as to why you are removing L925-L929 :). Without these lines virtualenvs in cygwin will always get the warning.

```python
In [1]: import os
   ...: import sys
   ...: import pathlib

In [2]: os.environ['VIRTUAL_ENV']
Out[2]: '/cygdrive/c/test_virtualenv_c'

In [3]: sys.executable
Out[3]: 'C:\\test_virtualenv_c\\Scripts\\python.exe'

In [4]: pathlib.Path(sys.executable).parents[1] == pathlib.Path(os.environ['VIRTUAL_ENV'])
Out[4]: False
```

Reply:
Oh now I understand! Thanks.
L925-L929 are actually important.
Instead of those lines, can we add the following snippet now?
```py
        # In Cygwin paths like "c:\..." and '\cygdrive\c\...' are possible
        if str(p_venv).startswith("\\cygdrive"):
            p_venv = "C:" / Path(str(p_venv)[11:])
```
We won't need the `else` part.

Reply:
Also, I have changed my fork's master to be in line with this PR. So we can re-open the old PR https://github.com/ipython/ipython/pull/13134 so that I can make changes. Or if this is the only change needed, maybe you can make this change on your branch. 
Do you know any better method? 🤔

Reply:
I don't think that GH will see it as new changes, so that PR is dead either way. Just push the changes to your branch and let me know when you're finished - I'll update the PR here to match your branch.

Reply:
Alright! I have pushed the changes to my fork's master. Commit - https://github.com/Mithil467/ipython/commit/77fa88629d6f7a1fb9e9358fcdcc0bfba68db0aa

Reply:
Attaching a few screenshots to help with the review.
#### On windows -
![image](https://user-images.githubusercontent.com/41439841/133953776-47ca0734-542b-43f8-85d5-6fda87a1a4f1.png)
![image](https://user-images.githubusercontent.com/41439841/133953854-fb0a97ce-b66f-4ff6-86f6-ea0c55baec61.png)
#### On cygwin -
![image](https://user-images.githubusercontent.com/41439841/133953928-a6a20794-fbad-4b40-9edb-afbfc9c4c24e.png)
![image](https://user-images.githubusercontent.com/41439841/133953985-7be5cb91-4000-4a3e-b046-4fa473bc12fe.png)

Reply:
This is great, exactly what I needed :)! Thanks!